Merge type definition from CRD to ApiServer and regenerated all the codes

medinatiger opened the pull request:
Due to the issue below, I have to temporarily remove custom type CloudStatusError and MachineStatusError.
https://github.com/kubernetes-incubator/apiserver-builder/issues/176

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kube-deploy/507)
<!-- Reviewable:end -->

medinatiger commented:
The code is verified by running
apiserver-boot run local
kubectl --kubeconfig kubeconfig api-versions

roberthbailey commented:
Can you squash into 2 commits? One for your manual changes (including changing the type from const -> string) and another for the code generation?

medinatiger committed:
Copy Cluster and Machine definition from CRD to ext-apiserver

There is some minior change due to the issue below.
https://github.com/kubernetes-incubator/apiserver-builder/issues/176

medinatiger committed:
Rerun Code Generation tool for the machine and cluster type change.

```sh
apiserver-boot run local
```

medinatiger commented:
@roberthbailey , squash them into 2 commit like you suggested, PTAL. Thanks!

roberthbailey commented:
/lgtm
